src/pages/storage/StorageBuckets.tsx

```diff
@@ -0,0 +1,310 @@
+import {
+  EmptyState,
+  Icon,
+  Row,
+  TablePagination,
+  useNotify,
+} from "@canonical/react-components";
+import Loader from "components/Loader";
+import ScrollableTable from "components/ScrollableTable";
+import StorageBucketsFilter, {
+  QUERY,
+  POOL,
+} from "pages/storage/StorageBucketsFilter";
+import { useDocs } from "context/useDocs";
+import {
+  ACTIONS_COL,
+  POOL_COL,
+  NAME_COL,
+  SIZE_COL,
+  URL_COL,
+} from "util/storageBucketTable";
+import useSortTableData from "util/useSortTableData";
+import CustomLayout from "components/CustomLayout";
+import PageHeader from "components/PageHeader";
+import HelpLink from "components/HelpLink";
+import NotificationRow from "components/NotificationRow";
+import { useLoadBuckets } from "context/useBuckets";
+import type { StorageBucketsFilterType } from "./StorageBucketsFilter";
+import StorageBucketsActions from "./actions/StorageBucketActions";
+import CreateBucketBtn from "./actions/CreateBucketBtn";
+import SelectableMainTable from "components/SelectableMainTable";
+import SelectedTableNotification from "components/SelectedTableNotification";
+import ResourceLink from "components/ResourceLink";
+import usePanelParams, { panels } from "util/usePanelParams";
+import CreateBucketPanel from "./panels/CreateBucketPanel";
+import StorageBucketBulkDelete from "./actions/StorageBucketBulkDelete";
+import type { LxdStorageBucket } from "types/storage";
+
+const StorageBuckets: FC = () => {
+  const docBaseLink = useDocs();
+  const notify = useNotify();
+  const { project } = useParams<{ project: string }>();
+  const [searchParams] = useSearchParams();
+
+  const [selectedNames, setSelectedNames] = useState<string[]>([]);
+  const [processingNames, setProcessingNames] = useState<string[]>([]);
+  const panelParams = usePanelParams();
+
+  const filters: StorageBucketsFilterType = {
+    queries: searchParams.getAll(QUERY),
+    pools: searchParams.getAll(POOL),
+  };
+
+  if (!project) {
+    return <>Missing project</>;
+  }
+
+  const { data: buckets = [], error, isLoading } = useLoadBuckets(project);
+
+  const getBucketKey = (bucket: LxdStorageBucket) => {
+    return `${bucket.name}-${bucket.pool}-${bucket.location || ""}`;
+  };
+
+  useEffect(() => {
+    const validKeys = new Set(buckets.map(getBucketKey));
+    const validSelections = selectedNames.filter((name) => validKeys.has(name));
+    if (validSelections.length !== selectedNames.length) {
+      setSelectedNames(validSelections);
+    }
+  }, [buckets]);
+
+  if (error) {
+    notify.failure("Loading storage buckets failed", error);
+  }
+
+  const headers = [
+    {
+      content: NAME_COL,
+      sortKey: "name",
+      className: "name",
+    },
+    {
+      content: POOL_COL,
+      sortKey: "pool",
+      className: "pool",
+    },
+    {
+      content: SIZE_COL,
+      sortKey: "size",
+      className: "size",
+    },
+    {
+      content: URL_COL,
```

Comment:
```suggestion
      content: URL_COL,
      sortKey: "s3_url",
```
Enable sorting by url